### ISSUE TITLE
[RFC] Add sales object to product

### ISSUE THREAD

**Issue by kadewu** [saleor/saleor#10817]
The RFC prepared based on the issue https://github.com/saleor/saleor/issues/9477

Idea is to add new type for `Sale` for `Product` to not open specific fields

```graphql
type SalesProductInfo{
    id: ID!
    metadata: [MetadataItem!]!
    name: String!
    type: String!
    startDate: DateTime! 
    endDate: DateTime
    createdAt: DateTime! "or `created` to maintain consistency with Sale type"
    updatedAt: DateTime!

    categories(
        before: String
        after: String
        first: Int
        last: Int
    ): CategoryCountableConnection

    "needed permission ALL_PRODUCTS_PERMISSIONS"
    products(
        before: String
        after: String
        first: Int
        last: Int
    ): ProductCountableConnection

    "needed permission ALL_PRODUCTS_PERMISSIONS"
    variants(
        before: String
        after: String
        first: Int
        last: Int
    ): ProductVariantCountableConnection

    "needed permission ALL_PRODUCTS_PERMISSIONS"
    collections(
        before: String
        after: String
        first: Int
        last: Int
    ): CollectionCountableConnection

    translation(
        languageCode: LanguageCodeEnum!
    ): SaleTranslation

    currency: String
    discount: Float
}

type SalesProductInfoCountableConnection {
    pageInfo: PageInfo!
    edges: [SalesProductInfoCountableEdge!]!
    totalCount: Int
}

type SalesProductInfoCountableEdge {
    node: SalesProductInfo!
    cursor: String!
}
```

Add new field
```graphql
type Product implements Node & ObjectWithMetadata {
    id: ID!
    ...

    sales(
        before: String
        after: String
        first: Int
        last: Int
    ): SalesProductInfoCountableConnection
} 
```

**Comment by IKarbowiak** [comment]
Could you reformat the RFC to keep to the pattern that was used [here](https://github.com/saleor/saleor/pull/10423) f.e? I mean add the general goal at the beginning, `General assumptions` part, and `API changes`.

**Comment by IKarbowiak** [comment]
Why cannot we use the existing `Sale` type instead of introducing the new type `SalesProductInfo`?

**Comment by aniav** [comment]
I think it is no longer relevant given the catalogue Promotions approach.
